Source/Core/Core/Config/InterfaceSettings.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/Config/InterfaceSettings.h"
+
+namespace Config
+{
+const ConfigInfo<bool> DEBUGGER_SHOW_REGISTERS({System::Main, "Interface",
```

Comment:
I should have brought this up in the previous review, but are interface settings really something that should belong in Core? This seems like something that should be handled solely by the UI code if possible.

Reply:
I was told that - when possible - I should always make UI Settings part of Core.

Reply:
Was there a reason given for that? It doesn't really make sense from an interface or library perspective, since different frontends can support different feature-sets. For example, the no-gui build now has to initialize variables it'll never use before being able to do anything. Ditto for the Android build, which doesn't even save its UI state to the Core.

Ideally the core shouldn't directly know, or be able to know, anything about UI-specific stuff, especially window/dialog statuses.

Reply:
See the discussion in the PR #6005.

Reply:
I'm with @lioncash on this. It doesn't make a lot of sense to me to put UI state / ephemeral settings in core. Especially with the new config layer stuff: what would it mean for a gameini or for netplay to set DEBUGGER_SHOW_REGISTERS? What's the use case?

@delroth made a good point in that other PR about having these settings respect portable.txt. In that case, we could tell QSettings to keep the settings local to the binary directory, and still have the logical separation of the setting types.

Reply:
I don't really care, using QSettings is simpler anyway.

Reply:
@delroth Sorry I misunderstood that then I guess. That was my attempt at integrating it into the config system.

Anyway I guess there are the following other ways of implementing it:
1. Moving the InterfaceSettings class to UICommon / DolphinQt2 (probably the latter)
2. Using QSettings
3. Parsing and Writing to the INI manually

I'd be fine with either 1 or 2 (or another option that I didn't list here).

3 seems pretty cumbersome and regarding some aspects even worse than the current solution.

So what should I do?

Reply:
My preference is to (1) but (2) is fine as long as it behaves like normal Dolphin settings: stored in a mostly human editable format in Dolphin's User/Config directory.

Reply:
FWIW, I'm leaning on (2), since QSettings likely integrates with the rest of Qt and its event system in a more convenient way.

> stored in a mostly human editable format in Dolphin's User/Config directory.

This should be relatively easy to do for the most part. It can be kicked to disk in ini format by doing.

```cpp
// Opening directly
QSettings settings{"/path/to/ini.ini", QSettings::IniFormat};

// ... or if the path needs to be determined later or set before creating instances.
// All inis created through QSettings will be placed in the specified directory.
QSettings::setPath(QSettings::IniFormat, QSettings::UserScope, "/directory/to/put/inis/in");
```

Reply:
I'm for (2), since it's just so darn convenient.

I'd also claim that it's **not** a requirement for these to be stored in a human-editable format. They are much closer to stored UI state rather than actual configuration: what's the use-case for going in and manually editing the log window geometry, for instance?

Reply:
Can you claim that of all the settings that you will store in QSettings, none of them will ever have any use case for human editing?

If not, then it should be human editable.

FYI with DolphinWX I have a set of configs for the debugger windows which I toggle through a script depending on what I'm doing :)